backend/lib/calculate/adequacy.py

```diff
@@ -150,11 +150,12 @@ def calculate_adequacies(
     logger.debug('Calculating adequacies for {} provider locations and {} service areas.'.format(
         len(locations), len(service_area_ids)))
 
-    # Remove duplicates and convert to lat-longtuples.
-    locations = list({
-        Point(**location)
-        for location in locations
-    })
+    location_mapping = collections.defaultdict(list)
+    for location in locations:
+        point_id = location.pop('id')
+        location_mapping[Point(**location)].append(point_id)
```

Comment:
One `Point` object could correspond to multiple `point_ids`? Seems like they're not very good identifiers then!

Reply:
We can also move toward having `locations` be dictionaries with an `id` field and something like `coordinates` representing the unique list of lat/long tuples, in which case this would become `coordinates_to_id_map`

Reply:
Yep we can discuss that. For now I thought Point was fine as a unique location identifier, but locations can host multiple providers, no?